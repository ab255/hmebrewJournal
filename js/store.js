const store = [
  {beerName: 'Green Bomber', beerType: 'Imperial IPA', brewDate: '4/28/2013'},
  {beerName: 'Red Dawn', beerType: 'American Amber Ale', brewDate: '3/26/2013'}
];

module.exports = store;
