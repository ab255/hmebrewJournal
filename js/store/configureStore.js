import { AsyncStorage } from 'react-native';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, autoRehydrate } from 'redux-persist';

import reducers from '../reducers/reducers';


const createBrewStore = applyMiddleware(thunk)(createStore);

const configureStore = () => {
  const store = autoRehydrate()(createBrewStore)(reducers);
  persistStore(store, {storage: AsyncStorage});
  return store;
}

export default configureStore;
