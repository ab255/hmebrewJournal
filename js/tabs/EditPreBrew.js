import React, {Component} from 'react';
import {
  ScrollView,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'

import store from '../store'

class EditPreBrew extends Component {
  constructor(props) {
    super(props);
    this.state = {
      beerName: this.props.route.beer.beerName || '',
      beerType: this.props.route.beer.beerType || '',
      ingredients: this.props.route.beer.ingredients || '',
      water: this.props.route.beer.water || '',
      postBrewIngredients: this.props.route.beer.postBrewIngredients || '',
      totalCosts: this.props.route.beer.totalCosts || '',
    }
  }

  editBrew = async () => {
    let uid = JSON.stringify(this.props.route.beer.uuid)
    let brew = JSON.stringify(this.state)
    try {
      await store.edit(uid, brew)
      this.props.navigator.popToTop()
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    return (
      <KeyboardAwareScrollView style={styles.container}>
        <View>
          <Text style={styles.title}>Beer Name:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(beerName) => this.setState({beerName})}
            value={this.state.beerName}
            placeholder='Beer Name'
            autoCapitalize='words'
          />
        </View>
        <View>
          <Text style={styles.title}>Beer Type:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(beerType) => this.setState({beerType})}
            value={this.state.beerType}
            placeholder='Beer Style'
            autoCapitalize='words'
          />
        </View>
        <View>
          <Text style={styles.title}>Ingredients:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(ingredients) => this.setState({ingredients})}
            value={this.state.ingredients}
            placeholder='Beer Style'
          />
        </View>
        <View>
          <Text style={styles.title}>Water Notes:</Text>
          <TextInput
            multiline={true}
            style={styles.brewInput}
            onChangeText={(water) => this.setState({water})}
            value={this.state.water}
            placeholder='Water Used'
          />
        </View>
        <View>
          <Text style={styles.title}>Post Brew Ingredients:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(postBrewIngredients) => this.setState({postBrewIngredients})}
            value={this.state.postBrewIngredients}
            placeholder='Add post brew ingredients'
          />
        </View>
        <View>
          <Text style={styles.title}>Total Cost:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(totalCosts) => this.setState({totalCosts})}
            value={this.state.totalCosts}
            placeholder='Cost in Dollars'
            keyboardType='numbers-and-punctuation'
          />
        </View>
        <TouchableHighlight
        onPress={this.editBrew.bind(this)}
        >
          <View style={styles.submitButton}>
            <Text style={styles.buttonText}>Submit</Text>
          </View>
        </TouchableHighlight>
      </KeyboardAwareScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#911F27',
    flex: 1,
    padding: 10,
    paddingTop: 0,
  },
  title: {
    color: '#F7D098',
    fontWeight: 'bold',
    marginBottom: 5,
  },
  brewInput: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    color: '#630A10',
    height: 30,
    paddingLeft: 5,
    marginBottom: 10,
  },
  ingredientsInput: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    color: '#630A10',
    paddingLeft: 5,
    height: 100,
    fontSize: 16,
    marginBottom: 10,
  },
  submitButton: {
    alignSelf: 'center',
    backgroundColor: '#F7D098',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    width: 300,
    height: 30,
    margin: 20,
    },
  buttonText: {
    color: '#630A10',
    fontSize: 16,
  },
})

export default EditPreBrew;
