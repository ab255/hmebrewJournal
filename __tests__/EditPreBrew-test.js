import 'react-native';
import React from 'react';
import EditPreBrew from '../js/tabs/EditPreBrew.js';
import store from '../js/mockStore.js'

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  let beer = store
  const tree = renderer.create(
    <EditPreBrew route={ { beer } } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
