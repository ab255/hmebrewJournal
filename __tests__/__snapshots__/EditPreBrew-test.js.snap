exports[`test renders correctly 1`] = `
<ScrollView
  contentInset={
    Object {
      "bottom": 49,
    }
  }
  enableAutoAutomaticScroll={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  onScroll={[Function]}
  scrollEventThrottle={8}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#911F27",
      "flex": 1,
      "padding": 10,
      "paddingTop": 0,
    }
  }>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Beer Name:
    </Text>
    <TextInput
      autoCapitalize="words"
      onChangeText={[Function]}
      placeholder="Beer Name"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="Green Bomber" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Beer Type:
    </Text>
    <TextInput
      autoCapitalize="words"
      onChangeText={[Function]}
      placeholder="Beer Style"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="Imperial IPA" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Ingredients:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Beer Style"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="adsfasdfasdfalskdjf;lsdjfkljalhsdkjfhaksjdhfkjasdhlajkhsdkljfhakjsdhfkjlahsdfkjahsldkjfhasdkjfhadskljfhllkajsdhfkjashfdkjhasdkjfhaksdjhfkjasdhalkjsdhfkjhasdlkjfhasdlkjfhakjsdhlkjajskdhfkljahsdkfjhasdkjhfkajdshfkjhadskjlfhkajsdhfkjahsdkjfhakjsdhfkjasdhfkjhasdkjfhaskjdhfkajsdhfkjhaskjdfhaskjdhfkjashfdkjhaskljdhakjsdhflkjadshlkfjhdklajfhaklsdjhfkjlashdklfjhasdkjfhakjsdhfkjasdhfkhaskjdhfkajshdfkjlasdh" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Water Notes:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Water Used"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Post Brew Ingredients:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Add post brew ingredients"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Total Cost:
    </Text>
    <TextInput
      keyboardType="numbers-and-punctuation"
      onChangeText={[Function]}
      placeholder="Cost in Dollars"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="$80.00" />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F7D098",
          "borderRadius": 5,
          "height": 30,
          "justifyContent": "center",
          "margin": 20,
          "width": 300,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
            "fontSize": 16,
          }
        }>
        Submit
      </Text>
    </View>
  </View>
</ScrollView>
`;
